core/adapters/http.go

```diff
@@ -46,7 +55,9 @@ func (hga *HTTPGet) GetURL() string {
 
 // GetRequest returns the HTTP request including query parameters and headers
 func (hga *HTTPGet) GetRequest() (*http.Request, error) {
-	request, err := http.NewRequest("GET", hga.GetURL(), nil)
+	ctx, cancel := context.WithTimeout(context.Background(), httpDefaultTimeout)
```

Comment:
If we set the timeout here, it will not reset between requests, right? So the timeout will be set to 15 seconds among all attempts in total?

Reply:
Ooooh you might be right about that. Good catch.